src/record.rs

```diff
@@ -1,47 +1,227 @@
-use chrono::Utc;
+use chrono::prelude::*;
 use std::collections::HashMap;
 use std::fmt;
 use uuid::Uuid;
 
 use crate::header::WarcHeader;
 use crate::Error as WarcError;
 
-/// A single WARC record parsed from a data stream.
+/// A single WARC record as parsed from a data stream.
+///
+/// It is guaranteed to be well-formed, but may not be valid according to the specification.
 #[derive(Clone, Debug, PartialEq)]
-pub struct Record {
-    /// The WARC standard version this record conforms to.
+pub struct RawRecord {
+    /// The WARC standard version this record reports conformance to.
     pub version: String,
-    /// A set of all headers that are part of this record.
-    pub headers: HashMap<crate::header::WarcHeader, Vec<u8>>,
+    /// All headers that are part of this record.
+    pub headers: HashMap<WarcHeader, Vec<u8>>,
     /// The data body of this record.
     pub body: Vec<u8>,
 }
 
+/// A builder for WARC records from data.
+pub struct RecordBuilder {
+    version: String,
+    headers: HashMap<WarcHeader, Vec<u8>>,
+    body: Vec<u8>,
+    record_date: Option<DateTime<Utc>>,
```

Comment:
```suggestion
    date: Option<DateTime<Utc>>,
```

Having `record_` on these fields has a little stutter. Thoughts on dropping the `record_`?

Reply:
This was not a strong preference in my part. IT was the outcome of these general naming rules in my head:

1. If a header is named "WARC-Foo", drop the "WARC" as it's redundant.
2. If a field has a WARC and a non-WARC version, such as "Content Type" and "WARC-Type", indicate it's the WARC one by spelling that out. This is an exception to rule 1.
3. If "WARC" would be spelled out, but the data structure implies it, use the name of the data structure instead.

Rule 1 would suggest I write `date`.
Rule 2 is hit, because are two different dates as I recall, so it would be `warc_date`.
Rule 3 then changed it to `record_date`.

Since I already wrote `record_id`, `record_date` didn't strike me as obnoxious.

Reply:
SGTM. Nothing show stopping right now.